Makefile

```diff
@@ -54,6 +54,10 @@ format-clang:
 format-swift:
 	swiftlint --fix
 
+format-swift-staged:
+	echo "Running swiftlint --fix on staged files"
+	swiftlint --fix $(shell git diff --cached --diff-filter=d --name-only | grep '\.swift$$')
```

Comment:
`h`: This doesn't seem to make a difference on my machine. I get almost the same log output as when using `make format-swift` and it takes the same amount of time.

Reply:
Did you had any swift file staged?
If no files are staged, it will run on all files

Reply:
Ah yes, then it works. Can we maybe skip this when there are no staged swift files? My PRs often don't touch swift files.

Reply:
Let me check that, I think the pre commit hook already does this

Reply:
yeah, the command shouldn't run in that case
<img width="559" alt="image" src="https://github.com/user-attachments/assets/210c9e1d-37f0-46da-86eb-2816c8962104" />